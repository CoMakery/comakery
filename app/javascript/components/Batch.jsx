import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Icon from './styleguide/Icon'
import ContentElement from './ContentElement'
import CurrencyAmount from './CurrencyAmount'
import Userpics from './Userpics'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.1);
  height: 90px;
`

const RightBorder = styled.div`
  width: 2px;
  height: 90px;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, .2);
  background-color: #5037f7;
  z-index: 10;
  position: absolute;
  margin-left: -1px;
`

const Image = styled.img`
  width: 65px;
  height: 65px;
  padding: 13px 15px 12px 17px;
`

const Info = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`

const Title = styled.div`
  display: flex;
  align-items: baseline;
  justify-content: space-between;
  padding-right: 15px;
`

const Name = styled.div`
  padding: 15px 15px 7px 15px;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-transform: capitalize;
`

const Details = styled.div`
  display: flex;
  padding: 7px 15px 11px 15px;
  width: 100%;
`

const Buttons = styled.div`
  display: flex;
  width: 45px;
  justify-content: space-between;
  margin-right: 30px;
  align-self: flex-end;
  margin-left: auto;
`

const StyledIcon = styled(Icon)`
  height: 15px;
`

class Batch extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Wrapper>
          <RightBorder />
          {this.props.batch.diagramUrl &&
            <Image src={this.props.batch.diagramUrl} alt="Batch Image" />
          }
          <Info>
            <Title>
              <Name>
                {this.props.batch.name}
              </Name>

              <CurrencyAmount
                amount={this.props.batch.totalAmount}
                currency={this.props.batch.currency}
                logoUrl={this.props.batch.currencyLogo}
              />
            </Title>

            <Details>
              <ContentElement title="progress">
                {`${this.props.batch.completedTasks} / ${this.props.batch.totalTasks} Completed`}
              </ContentElement>

              <ContentElement title="batch type">
                {this.props.batch.specialty || 'General'}
              </ContentElement>

              <ContentElement title="team members">
                <Userpics pics={this.props.batch.teamPics} limit={3} />
              </ContentElement>

              <Buttons>
                <a href={this.props.batch.editPath}>
                  <StyledIcon name="iconEdit.svg" />
                </a>
                <a rel="nofollow" data-method="delete" href={this.props.batch.destroyPath}>
                  <StyledIcon name="iconTrash.svg" />
                </a>
              </Buttons>
            </Details>
          </Info>
        </Wrapper>
      </React.Fragment>
    )
  }
}

Batch.propTypes = {
  batch: PropTypes.object
}
Batch.defaultProps = {
  batch: {
    diagramUrl: null
  }
}
export default Batch
