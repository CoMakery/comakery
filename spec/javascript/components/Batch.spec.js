import React from 'react'
import { mount } from 'enzyme'
import Batch from 'components/Batch'

describe('Batch', () => {
  it('renders component correctly', () => {
    const wrapper = mount(<Batch />)

    expect(wrapper).toMatchSnapshot()
  })
})
