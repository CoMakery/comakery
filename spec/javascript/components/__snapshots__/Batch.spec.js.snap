// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Batch renders component correctly 1`] = `
<Batch
  batch={
    Object {
      "diagramUrl": null,
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gZMcBi",
            "isStatic": true,
            "lastClassName": "bZMqXO",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.1);
  height: 90px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gZMcBi",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-gZMcBi bZMqXO"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gqjmRU",
                  "isStatic": true,
                  "lastClassName": "eAZYhG",
                  "rules": Array [
                    "
  width: 2px;
  height: 90px;
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, .2);
  background-color: #5037f7;
  z-index: 10;
  position: absolute;
  margin-left: -1px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-gqjmRU",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-gqjmRU eAZYhG"
            />
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-jTzLTM",
                  "isStatic": true,
                  "lastClassName": "cNhanF",
                  "rules": Array [
                    "
  display: flex;
  flex-direction: column;
  width: 100%;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-jTzLTM",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-jTzLTM cNhanF"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-fjdhpX",
                        "isStatic": true,
                        "lastClassName": "bVpSSP",
                        "rules": Array [
                          "
  display: flex;
  align-items: baseline;
  justify-content: space-between;
  padding-right: 15px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-fjdhpX",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-fjdhpX bVpSSP"
                  >
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-jzJRlG",
                              "isStatic": true,
                              "lastClassName": "gDoncv",
                              "rules": Array [
                                "
  padding: 15px 15px 7px 15px;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-transform: capitalize;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-jzJRlG",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-jzJRlG gDoncv"
                        />
                      </StyledComponent>
                    </styled.div>
                    <CurrencyAmount
                      amount=""
                      currency=""
                      logoUrl=""
                    >
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bxivhb",
                                "isStatic": true,
                                "lastClassName": "ljVujs",
                                "rules": Array [
                                  "
  display: flex;
  flex-direction: row;
  align-items: center;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 500;
  color: #4a4a4a;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bxivhb",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-bxivhb ljVujs"
                          >
                            <styled.div>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-ifAKCX",
                                      "isStatic": true,
                                      "lastClassName": "gATvmT",
                                      "rules": Array [
                                        "
  margin-right: 0.5em;
",
                                      ],
                                    },
                                    "displayName": "styled.div",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-ifAKCX",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="sc-ifAKCX gATvmT"
                                />
                              </StyledComponent>
                            </styled.div>
                            <styled.div>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-EHOje",
                                      "isStatic": true,
                                      "lastClassName": "etusMN",
                                      "rules": Array [
                                        "
  margin-right: 0.5em;
  text-transform: uppercase;
",
                                      ],
                                    },
                                    "displayName": "styled.div",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-EHOje",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="sc-EHOje etusMN"
                                >
                                  (No Token)
                                </div>
                              </StyledComponent>
                            </styled.div>
                          </div>
                        </StyledComponent>
                      </styled.div>
                    </CurrencyAmount>
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-cSHVUG",
                        "isStatic": true,
                        "lastClassName": "iPABgF",
                        "rules": Array [
                          "
  display: flex;
  padding: 7px 15px 11px 15px;
  width: 100%;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-cSHVUG",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-cSHVUG iPABgF"
                  >
                    <ContentElement
                      title="progress"
                    >
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bdVaJa",
                                "isStatic": true,
                                "lastClassName": "gPWqlF",
                                "rules": Array [
                                  "
  display: flex;
  flex-direction: column;
  padding-right: 40px;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bdVaJa",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-bdVaJa gPWqlF"
                          >
                            <styled.div>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-bwzfXH",
                                      "isStatic": true,
                                      "lastClassName": "ekDaoc",
                                      "rules": Array [
                                        "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
  text-transform: uppercase;
  margin-bottom: 3px;
",
                                      ],
                                    },
                                    "displayName": "styled.div",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-bwzfXH",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="sc-bwzfXH ekDaoc"
                                >
                                  progress
                                </div>
                              </StyledComponent>
                            </styled.div>
                            <styled.div>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-htpNat",
                                      "isStatic": true,
                                      "lastClassName": "vZQMM",
                                      "rules": Array [
                                        "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #4a4a4a;
",
                                      ],
                                    },
                                    "displayName": "styled.div",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-htpNat",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="sc-htpNat vZQMM"
                                >
                                  undefined / undefined Completed
                                </div>
                              </StyledComponent>
                            </styled.div>
                          </div>
                        </StyledComponent>
                      </styled.div>
                    </ContentElement>
                    <ContentElement
                      title="batch type"
                    >
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bdVaJa",
                                "isStatic": true,
                                "lastClassName": "gPWqlF",
                                "rules": Array [
                                  "
  display: flex;
  flex-direction: column;
  padding-right: 40px;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bdVaJa",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-bdVaJa gPWqlF"
                          >
                            <styled.div>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-bwzfXH",
                                      "isStatic": true,
                                      "lastClassName": "ekDaoc",
                                      "rules": Array [
                                        "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
  text-transform: uppercase;
  margin-bottom: 3px;
",
                                      ],
                                    },
                                    "displayName": "styled.div",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-bwzfXH",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="sc-bwzfXH ekDaoc"
                                >
                                  batch type
                                </div>
                              </StyledComponent>
                            </styled.div>
                            <styled.div>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-htpNat",
                                      "isStatic": true,
                                      "lastClassName": "vZQMM",
                                      "rules": Array [
                                        "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #4a4a4a;
",
                                      ],
                                    },
                                    "displayName": "styled.div",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-htpNat",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="sc-htpNat vZQMM"
                                >
                                  General
                                </div>
                              </StyledComponent>
                            </styled.div>
                          </div>
                        </StyledComponent>
                      </styled.div>
                    </ContentElement>
                    <ContentElement
                      title="team members"
                    >
                      <styled.div>
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "sc-bdVaJa",
                                "isStatic": true,
                                "lastClassName": "gPWqlF",
                                "rules": Array [
                                  "
  display: flex;
  flex-direction: column;
  padding-right: 40px;
",
                                ],
                              },
                              "displayName": "styled.div",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "sc-bdVaJa",
                              "target": "div",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                        >
                          <div
                            className="sc-bdVaJa gPWqlF"
                          >
                            <styled.div>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-bwzfXH",
                                      "isStatic": true,
                                      "lastClassName": "ekDaoc",
                                      "rules": Array [
                                        "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
  text-transform: uppercase;
  margin-bottom: 3px;
",
                                      ],
                                    },
                                    "displayName": "styled.div",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-bwzfXH",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="sc-bwzfXH ekDaoc"
                                >
                                  team members
                                </div>
                              </StyledComponent>
                            </styled.div>
                            <styled.div>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-htpNat",
                                      "isStatic": true,
                                      "lastClassName": "vZQMM",
                                      "rules": Array [
                                        "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #4a4a4a;
",
                                      ],
                                    },
                                    "displayName": "styled.div",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-htpNat",
                                    "target": "div",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <div
                                  className="sc-htpNat vZQMM"
                                >
                                  <Userpics
                                    limit={3}
                                    pics={Array []}
                                  >
                                    <styled.div>
                                      <StyledComponent
                                        forwardedComponent={
                                          Object {
                                            "$$typeof": Symbol(react.forward_ref),
                                            "attrs": Array [],
                                            "componentStyle": ComponentStyle {
                                              "componentId": "sc-gzVnrw",
                                              "isStatic": true,
                                              "lastClassName": "dDxFqh",
                                              "rules": Array [
                                                "
  display: flex;
  flex-direction: row;
  align-items: center;
",
                                              ],
                                            },
                                            "displayName": "styled.div",
                                            "foldedComponentIds": Array [],
                                            "render": [Function],
                                            "styledComponentId": "sc-gzVnrw",
                                            "target": "div",
                                            "toString": [Function],
                                            "warnTooManyClasses": [Function],
                                            "withComponent": [Function],
                                          }
                                        }
                                        forwardedRef={null}
                                      >
                                        <div
                                          className="sc-gzVnrw dDxFqh"
                                        >
                                          <styled.div>
                                            <StyledComponent
                                              forwardedComponent={
                                                Object {
                                                  "$$typeof": Symbol(react.forward_ref),
                                                  "attrs": Array [],
                                                  "componentStyle": ComponentStyle {
                                                    "componentId": "sc-htoDjs",
                                                    "isStatic": true,
                                                    "lastClassName": "jYtsma",
                                                    "rules": Array [
                                                      "
  margin-right: 8px;
",
                                                    ],
                                                  },
                                                  "displayName": "styled.div",
                                                  "foldedComponentIds": Array [],
                                                  "render": [Function],
                                                  "styledComponentId": "sc-htoDjs",
                                                  "target": "div",
                                                  "toString": [Function],
                                                  "warnTooManyClasses": [Function],
                                                  "withComponent": [Function],
                                                }
                                              }
                                              forwardedRef={null}
                                            >
                                              <div
                                                className="sc-htoDjs jYtsma"
                                              />
                                            </StyledComponent>
                                          </styled.div>
                                        </div>
                                      </StyledComponent>
                                    </styled.div>
                                  </Userpics>
                                </div>
                              </StyledComponent>
                            </styled.div>
                          </div>
                        </StyledComponent>
                      </styled.div>
                    </ContentElement>
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-kAzzGY",
                              "isStatic": true,
                              "lastClassName": "hyUmFa",
                              "rules": Array [
                                "
  display: flex;
  width: 45px;
  justify-content: space-between;
  margin-right: 30px;
  align-self: flex-end;
  margin-left: auto;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-kAzzGY",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-kAzzGY hyUmFa"
                        >
                          <a>
                            <Styled(Icon)
                              name="iconEdit.svg"
                            >
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-chPdSV",
                                      "isStatic": true,
                                      "lastClassName": "DlMfj",
                                      "rules": Array [
                                        "
  height: 15px;
",
                                      ],
                                    },
                                    "displayName": "Styled(Icon)",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-chPdSV",
                                    "target": [Function],
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                                name="iconEdit.svg"
                              >
                                <Icon
                                  className="sc-chPdSV DlMfj"
                                  name="iconEdit.svg"
                                >
                                  <img
                                    className="icon icon__iconEdit-svg sc-chPdSV DlMfj"
                                    src="test-file-stub"
                                  />
                                </Icon>
                              </StyledComponent>
                            </Styled(Icon)>
                          </a>
                          <a
                            data-method="delete"
                            rel="nofollow"
                          >
                            <Styled(Icon)
                              name="iconTrash.svg"
                            >
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-chPdSV",
                                      "isStatic": true,
                                      "lastClassName": "DlMfj",
                                      "rules": Array [
                                        "
  height: 15px;
",
                                      ],
                                    },
                                    "displayName": "Styled(Icon)",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-chPdSV",
                                    "target": [Function],
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                                name="iconTrash.svg"
                              >
                                <Icon
                                  className="sc-chPdSV DlMfj"
                                  name="iconTrash.svg"
                                >
                                  <img
                                    className="icon icon__iconTrash-svg sc-chPdSV DlMfj"
                                    src="test-file-stub"
                                  />
                                </Icon>
                              </StyledComponent>
                            </Styled(Icon)>
                          </a>
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Batch>
`;
